prow/plugins/lgtm/lgtm.go

```diff
@@ -176,7 +176,7 @@ func handlePullRequestReviewEvent(pc plugins.Agent, e github.ReviewEvent) error
 	return handlePullRequestReview(pc.GitHubClient, pc.PluginConfig, pc.OwnersClient, pc.Logger, cp, e)
 }
 
-func handleGenericComment(gc githubClient, config *plugins.Configuration, ownersClient repoowners.Interface, log *logrus.Entry, cp commentPruner, e github.GenericCommentEvent) error {
+func handleGenericComment(gc githubClient, config *plugins.Configuration, ownersClient repoowners.Interface, log *logrus.Entry, cp commentPruner, e github.GenericCommentEvent) plugins.Status {
```

Comment:
errors are always returned as distinct values and not as part of a struct. IMHO it would also make sense to use a `tookAction` boolean, because allowing an arbitrary string means that plugins will likely invent all kind of possible descriptions of what they did or did not do

Reply:
Also not a fan of calling it `ExternalApiCall` because I assume there are plugins that make api calls to figure out if they need to do something. Maybe the definition of `tookAction` would be "Mutated any external state"?

Reply:
Regarding errors you are right, I'm coming from the C++ world and there was no multiple returns... I have to get used to it, thanks.
`tookAction` is a fine name. I'll just rename it. I would still keep it in a struct, even if there is only one value to have it extendable in the future. What do you think?
I am not exactly sure what you mean by "allowing an arbitrary string". I think map params coded as labels for Prometheus have to be strings. We can have "true" value, "false" value but we should also indicate that param is unused or unknown (for the "classic" version of handlers). Returning the "false" string value would not be correct, as actually, we do not know whether `tookAction` happened.
Alternative to that, instead of extending the interface by another function version would actually be changing the interface, but that will affect the whole plugin infrastructure.

Reply:
> tookAction is a fine name. I'll just rename it. I would still keep it in a struct, even if there is only one value to have it extendable in the future. What do you think?

Personally I think using a struct for extensibility is a good idea

Reply:
Yes, the struct is fine (modulo the embedded error). What confused me is how this gets mapped to the prom label value. Could we maybe just set that to the empty string if we don't know and true/false otherwise?

Reply:
Renamed to `tookAction` and moved `error` out of the struct. Also, reported empty string and true/false otherwise.